modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -1133,6 +1145,8 @@ private void updateGroupMetrics() {
         // TODO: IGNITE-11330: Update metrics for touched cache only.
         // Due to historical rebalancing "EstimatedRebalancingKeys" metric is currently calculated for the whole cache
         // group (by partition counters), so "RebalancedKeys" and "RebalancingKeysRate" is calculated in the same way.
+        grp.metrics().onRebalanceKeyReceived();
```

Comment:
I don't think that this is the right place for incrementing of receiving keys. Let's call this outside the `updateGroupMetrics` since this is a method-candidate to remove.